nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -140,6 +141,50 @@ in
         };
       };
 
+      services.nginx = mkIf cfg.nginx.enable {
+        enable = mkDefault true;
+        virtualHosts."${cfg.settings.hostname}" = {
```

Comment:
I was looking for this feature, and my use case involves `forceSSL = true` and `enableACME = true` in the virtual host configuration for lemmy.

I am not knowledgeable with Nix but would it be possible with this code?

Reply:
yes, most services in nixpkgs leave ACME/SSL up to the user. a config with those enabled would look like this:

```nix
services.lemmy = {
  enable = true;
  settings.hostname = "lemmy.example.com";
  nginx.enable = true;
};

# `nginx.enable` above creates a "lemmy.example.com" virtualhost, and you can add to it like so:
services.nginx.virtualHosts."lemmy.example.com" = {
  forceSSL = true;
  enableACME = true;
};
```

"option definition merging" and "nixos module system" would be the topics to study if you're curious how the above works under the hood (i don't have specific resources for that handy, but [here's](https://nixos.wiki/wiki/Support) usually where i start).